Zend/zend_vm_execute.skl

```diff
@@ -13,6 +13,15 @@ ZEND_API void {%EXECUTOR_NAME%}_ex(zend_execute_data *ex)
 
 	{%INTERNAL_LABELS%}
 
+	/* Force C compiler to store preserved registers to allow JIT using them */
```

Comment:
> Does this not make problems with MSVC or non-GNU-compatible C compilers? (e.g. intel ICC)?

MSVC is not affected by this macros an JIT produces a less efficient code.
ICC is actually based on GCC.

Reply:
> I understand you don't save esi and edi in i386 for example because they are global registers. 

``esi`` and ``edi`` are saved/restored manually through ``orig_opline`` ans ``orig_execute_data``.
 
> What if GCC global regs is not enabled?

e.g. for CLANG build...

> Don't you then need to save those registers too?

Good catch. The current JIT implementation uses this feature only for HYBRID VM.
It probably makes sense to wrap this builtins with ``#if```.

For CALL VM, JIT now saves/restores all the preserved registers, but this may be improved, at least for CLANG.